code/src/components/MainGame.js

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+
+import styled from "styled-components";
+import { useWindowSize } from "@react-hook/window-size";
+
+import { useSelector } from "react-redux";
+
+import GameStage from "./GameStage";
+import Winner from "./Winner";
+import Compass from "./Compass";
+
+import ceiling from "../assets/ceiling.png";
+import cave from "../assets/cave.png";
+import last from "../assets/last.png";
+import color from "../assets/color.png";
+import paper from "../assets/paper.png";
+import steampunk from "../assets/steampunk.png";
+import colorfulcave from "../assets/colorfulcave.png";
+import back from "../assets/back.gif";
+
+import TypeWriterEffect from "react-typewriter-effect";
+
+const GameBox = styled.div`
+  height: 150vh;
+  width: 100vw;
+  background-image: url(${(props) => props.bg});
+  background-repeat: no-repeat;
+  background-position: center;
+  background-size: cover;
+  background-attachment: fixed;
+  background-color: black;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  color: white;
+
+  @media (min-width: 768px) {
+    height: 100vh;
+  }
+
+  .compassBox {
+    position: absolute;
+    top: 3%;
+    left: 10%;
+
+    @media (min-width: 768px) {
+      top: 10%;
+    }
+  }
+
+  .typeBox {
+    margin-bottom: 20rem;
+    z-index: 5;
+  }
+
+  .react-typewriter-text-wrap {
+    position: absolute;
+    top: 14%;
+    width: 70%;
+
+    @media (min-width: 768px) {
+      top: 20%;
+    }
+
+    .react-typewriter-text {
+      font-size: 23px;
+
+      @media (min-width: 768px) {
+        font-size: 2em;
+      }
+    }
+  }
+`;
+
+const DescriptionBox = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 70%;
+  height: auto;
+  padding: 2em;
+  align-items: center;
+  border-radius: 20px;
+  justify-content: center;
+`;
+
+export const MainGame = () => {
+  const bgSwitcher = (coordinates) => {
```

Comment:
We did the backgrounds as an object, if you are interested in seeing an alternative way of doing it to the switch. Like this:

```
export const LocationBackground = ({children, coordinates}) => {

    const backgrounds = {
        "0,0": "./assets/archway-cavern.png",
        "1,0": "./assets/light-rays-cavern.jpg",
        "1,1": "./assets/brighter-cavern.jpg",
        "0,1": "./assets/ice-cavern.jpg",
        "0,2": "./assets/ice-cavern.jpg",
        "0,3": "./assets/bright-light.jpg",
        "1,3": "./assets/calm.jpg",
    }
    return (
        <BackgroundImage image={backgrounds[coordinates]}>
            {children}
        </BackgroundImage>
    );
};
```

(The children is just because it's a component that we're mounting with other stuff inside it too, you can ignore that bit, and BackgroundImage is a styled component that takes the image as props)